### PR TITLE
Add sentry release for connectors

### PR DESCRIPTION
**Pull request by tuliren**
## Summary
- Create a Sentry release whenever a new connector version is published.
- In this way, we can track an exception back to the version and suspicious commit.
- Sentry doc
  - https://docs.sentry.io/product/releases/suspect-commits/
  - https://docs.sentry.io/product/cli/releases/
- Relate to https://github.com/airbytehq/airbyte/issues/9104.

### CODE REVIEW

File: .github/workflows/publish-command.yml [hunk 1]
```diff
@@ -94,15 +94,39 @@ jobs:
           ci_credentials ${{ github.event.inputs.connector }}
         env:
           GCP_GSM_CREDENTIALS: ${{ secrets.GCP_GSM_CREDENTIALS }}
-      - run: |
+      - name: Prepare Sentry Release
+        if: startsWith(github.event.inputs.connector, 'connectors')
+        run: |
+          curl -sL https://sentry.io/get-cli/ | bash
```
Comment (edgao) [inline .github/workflows/publish-command.yml:100]: does it make sense to commit this file to git? Then if e.g. sentry.io goes down, our ability to release isn't impacted

**Review by edgao — COMMENTED** [review]
I'm not familiar with how sentry works, is this basically doing:
1. install/setup sentry tooling
2. create a release with name like `source-whatever@1.2.3` and associate it with a range of git commits (?)
3. do the normal connector publish stuff
4. finish the sentry release

and (presumably) if any of the sentry steps fail, then the entire release fails? (which seems like a good thing, just want to confirm)

**Comment by tuliren** [comment]
> create a release with name like source-whatever@1.2.3 and associate it with a range of git commits (?)

Yes. I have installed the Sentry integration for Github. So Sentry can automatically associate the new commits for a new release. One flaw in the current process is that we typically publish a connector in a PR, and do a squash merge. So some of the PR commits will be included in the release, which won't exist in the master branch after the PR is merged. However, I cannot think of a quick way to solve this problem.

> if any of the sentry steps fail, then the entire release fails?

I think Sentry is flexible most of the time. As long as it is installed correctly, it won't fail if there is an issue.

> does it make sense to commit this file to git? Then if e.g. sentry.io goes down, our ability to release isn't impacted

I'd prefer not to. If we commit it, it becomes an implicit dependency we need to maintain. Sentry is usually pretty robust. We will re-evaluate if there are frequent failures.

**Pull request merged**